Sources/Build/misc.swift

```diff
@@ -173,3 +173,41 @@ extension Product {
         return ((), plist: s)
     }
 }
+
+extension SystemPackageProvider {
+    
+    var installText: String {
+        switch self {
+        case .Brew(let name):
+            return "    brew install \(name)\n"
+        case .Apt(let name):
+            return "    apt-get install \(name)\n"
+        }
+    }
+    
+    static func providerForCurrentPlatform(providers: [SystemPackageProvider]) -> SystemPackageProvider? {
+        guard let uname = try? popen(["uname"]).chomp().lowercased() else { return nil }
```

Comment:
It would be good to factor out the information about the target platform somewhere else (in particular, just using a `popen(["uname"])` here to infer the platform is gross).

Also, I think it probably make more sense to just have `SystemPackageProvider` have a method which determines whether or not it is appropriate for the build environment, and then have the loop here be simply:
```
for provider in providers {
  if provider.isAvailable {
    return provider
  }
}
return nil
```